alphagov/paas-docker-cloudfoundry-tools
# [#150711834] Add a new bosh-shell Docker image

## What

https://www.pivotaltracker.com/story/show/150711834

Add a new bosh-shell Docker image

This is intended for use as an interactive shell by an operator on their local machine. This differs from the bosh-cli, which is intended for use in our pipelines.

## How to review

Review in conjunction with https://github.com/alphagov/paas-bootstrap/pull/100
If you don't you will not have the script for testing the container and your Bosh won't have the intended security groups.

* Make sure the latest build of this image in Docker Hub is completed: 
  https://hub.docker.com/r/governmentpaas/bosh-shell/builds/
  It should have a tag corresponding to this branch name (direct-ssh-bosh).
* Use `DEPLOY_ENV=dev make bosh-cli` on the [dev branch of paas-bootstrap](https://github.com/alphagov/paas-bootstrap/tree/direct-ssh-bosh) to run the container.
* Check you can perform all Bosh operations:
  * Basic commands: `bosh vms`
  * Commands against a VM: `bosh do-something vm/index`
  * SSH: `bosh-ssh vm/index`. **Note:** we have added a `bosh-ssh` wrapper script to avoid you having to pass in gateway VM details (to SSH via the Bosh VM).
* Code review
* Travis should take care of the tests. You can refer to the README in the root directory for running them yourself locally.
* Check the way we handle secrets doesn't expose anything it shouldn't.


## Who

Anyone but me or @camelpunch
